EW-995 Add new tsp rest client (#5236)
hpi-schul-cloud/schulcloud-server: apps/server/src/infra/tsp-client/README.md

# TSP API CLIENT

> A short introduction how this module can be used and the api client is generated.

## How to use the api client

The clients for the different Tsp endpoints should be created through TspClientFactory.
Through the create methods of the factory the basic configuration will be set. Currently the
factory sets the base url and generates the JWT used for the requests. You can use the client
like this:

```typescript
export class MyNewService {
    // inject the factory into the constructor
    constructor(private readonly tspClientFactory: TspClientFactory) {}

    public async doSomeStuff(): Promise<void> {
        // this will create a fully initialized client
        const exportClient = tspClientFactory.createExportClient();

        // calling the api
        const versionResponse = await exportClient.version();

        // do other stuff...
    }
}
```

## How the code generation works

We are using the openapi-generator-cli to generate apis, models and supporting files in the

> `generated` directory. **DO NOT** modify anything in the `generated` folder, because it will
> deleted on the next client generation.

Reviewer: ```suggestion
it will be deleted
```

> `generated` directory. **DO NOT** modify anything in the `generated` folder, because it will
> deleted on the next client generation.

Reviewer: ```suggestion
because it will be deleted on the next client generation.
```

Reviewer 2: Done
